java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -120,7 +121,7 @@ public String toString() {
     if (resultConstraint != null) {
       stateStream = stateStream.map(s -> s.addConstraint(sv, resultConstraint));
     }
-    return stateStream.distinct();
+    return stateStream.distinct().map(ps -> new Pair<>(this, ps));
```

Comment:
This is a rather complicated hack to try to keep using stream api against all odds in EGWalker. I would try to get rid of Pair class and do a proper old loop.